active_learning/liver_tumor_al/active_learning.py

```diff
@@ -0,0 +1,610 @@
+            most_unstable_names = scores_keys[most_unstable]
+            most_unstable_names = most_unstable_names.tolist()
+
+            rem_indices = sorted_indices[: -args.queries]
+            rem_names = scores_keys[rem_indices]
+            rem_names = rem_names.tolist()
+
+            copy_unl_d = np.array(copy_unl_d)
+
+            # Form the new JSON
+            # Get indices from unlabeled data pool using most unstable names
+            grab_indices = []
+            for each_unstable_name in most_unstable_names:
+                for idx_unl, each_sample in enumerate(copy_unl_d):
+                    print(each_sample)
+                    if each_unstable_name == each_sample["image"]:
+                        grab_indices.append(idx_unl)
+
+            samples = copy_unl_d[grab_indices]
+            for each in samples:
+                copy_train_d.append(each)
+
+            grab_rem = []
+            unlabeled_data = []
+            for each_rem_name in rem_names:
+                for idx_unl, each_sample in enumerate(copy_unl_d):
+                    if each_rem_name == each_sample["image"]:
+                        print(each_sample)
+                        grab_rem.append(idx_unl)
+
+            rem_samples = copy_unl_d[grab_rem]
+            for each in rem_samples:
+                unlabeled_data.append(each)
+            copy_unl_d = unlabeled_data
+
+            print("Updated Json File Sample Count")
+            print("Number of Training samples for next iter: {}".format(len(copy_train_d)))
+            print("Number of Unlabeled samples for next iter: {}".format(len(copy_unl_d)))
+
+        elif args.strategy == "random":
+            print("Random Strategy is being used for selection")
+            grab_indices = random.sample(range(0, len(copy_unl_d)), args.queries)
+
+            copy_unl_d = np.array(copy_unl_d)
+            samples = copy_unl_d[grab_indices]
+            for each in samples:
+                copy_train_d.append(each)
+
+            copy_unl_d = np.delete(copy_unl_d, grab_indices)
+            copy_unl_d = copy_unl_d.tolist()
+
+            print("Updated Json File Sample Count")
+            print("Number of Training samples for next iter: {}".format(len(copy_train_d)))
+            print("Number of Unlabeled samples for next iter: {}".format(len(copy_unl_d)))
+
+        # Write new json file
+        new_json_dict = {}
+        new_json_dict["training"] = copy_train_d
+        new_json_dict["unlabeled"] = copy_unl_d
+        new_json_dict["validation"] = copy_val_d
+        new_json_dict["test"] = copy_test_d
+
+        new_json_file_path = os.path.join(json_base_dir, "json_iter_{}.json".format(active_iter))
+        with open(new_json_file_path, "w") as j_file:
+            json.dump(new_json_dict, j_file)
+        j_file.close()
+
+        # Update New Json path
+        new_json_path = new_json_file_path
+        print("Active Iteration {} Completed".format(active_iter))
+
+    return None
```

Comment:
This seems unnecessary as the `main` function need not to return anything typically.

Reply:
removed.